CMakeLists.txt

```diff
@@ -367,24 +372,93 @@ else()
+
+  #Cortex-A5/9  -mfpu=neon-fp16
+  #Cortex-A7/15 -mfpu=neon-vfpv4
+  #Cortex-A8    -mfpu=neon
+  #ARMv8	-FP and SIMD on by default for all ARM8v-a series, NO -mfpu setting needed
+
+  #For custom -mtune, processor IDs for ARMv8-A series:
+  #0xd04 - Cortex-A35
+  #0xd07 - Cortex-A57
+  #0xd08 - Cortex-A72
+  #0xd03 - Cortex-A73
+
+  if(NOT ARM8)
+  CHECK_CXX_ACCEPTS_FLAG(-mfpu=vfp3-d16 CXX_ACCEPTS_VFP3_D16)
+  CHECK_CXX_ACCEPTS_FLAG(-mfpu=vfp4 CXX_ACCEPTS_VFP4)
+  CHECK_CXX_ACCEPTS_FLAG(-mfloat-abi=hard CXX_ACCEPTS_MFLOAT_HARD)
+  CHECK_CXX_ACCEPTS_FLAG(-mfloat-abi=softfp CXX_ACCEPTS_MFLOAT_SOFTFP)
+  endif()
+
+  CHECK_CXX_ACCEPTS_FLAG(-mfix-cortex-a53-835769 CXX_ACCEPTS_MFIX_CORTEX_A53_835769)
+  CHECK_CXX_ACCEPTS_FLAG(-mfix-cortex-a53-843419 CXX_ACCEPTS_MFIX_CORTEX_A53_843419)
+  CHECK_CXX_ACCEPTS_FLAG(-mtune=native CXX_ACCEPTS_MTUNE_NATIVE)
+
   if(ARM6)
-    message(STATUS "Setting ARM6 C and C++ flags")
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mfpu=vfp -mfloat-abi=hard")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mfpu=vfp -mfloat-abi=hard")
+    message(STATUS "Selecting VFP")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mfpu=vfp")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mfpu=vfp")
   endif()
 
   if(ARM7)
-    message(STATUS "Setting ARM7 C and C++ flags")
+  if(CXX_ACCEPTS_VFP3_D16 AND NOT CXX_ACCEPTS_VFP4)
+    message(STATUS "Selecting VFP3")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mfpu=vfp3-d16")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mfpu=vfp3-d16")
+  endif()
+
+  if(CXX_ACCEPTS_VFP4)
+    message(STATUS "Selecting VFP4")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mfpu=vfp4")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mfpu=vfp4")
+  endif()
+
+  if(CXX_ACCEPTS_MFLOAT_HARD)
+    message(STATUS "Setting Hardware ABI for Floating Point")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mfloat-abi=hard")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mfloat-abi=hard")
   endif()
 
+  if(CXX_ACCEPTS_MFLOAT_SOFTFP AND NOT CXX_ACCEPTS_MFLOAT_HARD)
+    message(STATUS "Setting Software ABI for Floating Point")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mfloat-abi=softfp")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mfloat-abi=softfp")
+  endif()
+  endif(ARM7)
+
+  if(CXX_ACCEPTS_MFIX_CORTEX_A53_835769)
+    message(STATUS "Enabling Cortex-A53 workaround 835769")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mfix-cortex-a53-835769")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mfix-cortex-a53-835769")
+  endif()
+
+  if(CXX_ACCEPTS_MFIX_CORTEX_A53_843419)
+    message(STATUS "Enabling Cortex-A53 workaround 843419")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mfix-cortex-a53-843419")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mfix-cortex-a53-843419")
+  endif()
+
+  if(CXX_ACCEPTS_MTUNE_NATIVE)
+    message(STATUS "Tuning for Native ARM Architecture")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mtune=native")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mtune=native")
+  endif()
```

Comment:
We set -march=native in default build. I vote for dropping -mtune. In deployment builds without -march=native, not sure if it's worth tuning it for the build host, especially since @fluffypony builds in QEMU. Sorry to oscillate on this.